edb/lib/cal.edgeql

```diff
@@ -1920,3 +1922,160 @@ std::max(vals: SET OF array<cal::date_duration>) -> OPTIONAL array<cal::date_dur
+            THEN edgedb.local_datetime_range_t("lower", "upper", '[]')
+            WHEN NOT "inclower" AND "incupper"
+            THEN edgedb.local_datetime_range_t("lower", "upper", '(]')
+            WHEN NOT "inclower" AND NOT "incupper"
+            THEN edgedb.local_datetime_range_t("lower", "upper", '()')
+            WHEN "inclower" AND NOT "incupper"
+            THEN edgedb.local_datetime_range_t("lower", "upper", '[)')
+        END
+    $$;
+};
+
+
+CREATE FUNCTION
+std::range(
+    lower: optional cal::local_date = {},
+    upper: optional cal::local_date = {},
+    named only inclower: bool = true,
+    named only incupper: bool = false
+) -> range<cal::local_date>
+{
+    SET volatility := 'Immutable';
+    USING SQL $$
+        SELECT CASE
+            WHEN "inclower" AND "incupper"
+            THEN daterange("lower", "upper", '[]')
+            WHEN NOT "inclower" AND "incupper"
+            THEN daterange("lower", "upper", '(]')
+            WHEN NOT "inclower" AND NOT "incupper"
+            THEN daterange("lower", "upper", '()')
+            WHEN "inclower" AND NOT "incupper"
+            THEN daterange("lower", "upper", '[)')
+        END
+    $$;
+};
+
+
+CREATE FUNCTION
+std::range_unpack(
+    val: range<cal::local_datetime>,
+    step: cal::relative_duration
+) -> set of cal::local_datetime
+{
+    SET volatility := 'Immutable';
+    USING SQL $$
+        SELECT
+            generate_series(
+                (
+                    lower(val) + (
+                        CASE WHEN lower_inc(val)
+                            THEN '0'::interval
+                            ELSE step
+                        END
+                    )
+                )::timestamp,
+                (
+                    upper(val) - (
+                        CASE WHEN upper_inc(val)
+                            THEN '0'::interval
+                            ELSE step
+                        END
+                    )
+                )::timestamp,
+                step::interval
+            )::edgedb.timestamp_t
+    $$;
+};
+
+
+CREATE FUNCTION
+std::range_unpack(
+    val: range<cal::local_date>
+) -> set of cal::local_date
+{
+    SET volatility := 'Immutable';
+    USING SQL $$
+        SELECT
+            generate_series(
+                (
+                    lower(val) + (
+                        CASE WHEN lower_inc(val)
+                            THEN '0'::interval
+                            ELSE 'P1D'::interval
+                        END
+                    )
+                )::timestamp,
+                (
+                    upper(val) - (
+                        CASE WHEN upper_inc(val)
+                            THEN '0'::interval
+                            ELSE 'P1D'::interval
+                        END
+                    )
+                )::timestamp,
+                'P1D'::interval
+            )::edgedb.date_t
+    $$;
+};
+
+
+CREATE FUNCTION
+std::range_unpack(
+    val: range<cal::local_date>,
```

Comment:
We should also have functions to unpack ranges for ints/floats/etc?

Reply:
We do that in `31-rangefuncs.edgeql`.